src/test/shell/bazel/bazel_cc_code_coverage_test.sh

```diff
@@ -143,46 +165,68 @@ function run_coverage() {
 
 function test_cc_test_coverage() {
   run_coverage > "$TEST_log"
+  assert_coverage_result
+}
 
-  # After running the test in coverage_srcs/t.cc, the sources covered are the
-  # test itself and the source file a.cc.
-  # For more details about the lcov format see
-  # http://ltp.sourceforge.net/coverage/lcov/geninfo.1.php
-  # The expected result can be constructed manually by following the lcov
-  # documentation and manually checking what lines of code are covered when
-  # running the test.
-  cat <<EOF > expected_result.dat
-TN:
-SF:coverage_srcs/a.cc
-FN:3,_Z1ab
-FNDA:1,_Z1ab
-FNF:1
-FNH:1
-DA:3,1
-DA:4,1
-DA:5,1
-DA:7,0
-LF:4
-LH:3
-end_of_record
-TN:
-SF:coverage_srcs/t.cc
-FN:4,main
-FNDA:1,main
-FNF:1
-FNH:1
-DA:4,1
-DA:5,1
-DA:6,1
-LF:3
-LH:3
-end_of_record
-EOF
-
-  # tools/test/collect_cc_coverage.sh places the coverage result in
-  # $COVERAGE_OUTPUT_FILE
-  diff -u expected_result.dat "$COVERAGE_OUTPUT_FILE_VAR" >> "$TEST_log" \
```

Comment:
Why wasn't the diff enough? I find the new version way more complicated.

Reply:
Yes it is more complicated and I hoped I could avoid this. When I initially ran the test on my Ubuntu laptop it failed because the order of the source files was inverted. The lcov documentation says nothing about the order of the source files in the coverage report, so no order is guaranteed. If no order is guaranteed, even if the test wouldn't have failed, it is still the right thing to do.